### PR TITLE
Please review improvements

### PR DESCRIPTION
**Pull request by kristianmandrup**
- customizable via config object/file, better docs, more granular design
- added args to some of the fakers and added a few extras such as: geo coordinate near
- allow custom faker sections and split into separate faker function section files
- allow passing custom faker object via config

PS: Not yet tested

### CODE REVIEW

File: src/fake/functions/internet.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ export function internetFunctions(fakerOpts: any = {}) {
     },
     macAddress: () => faker.internet.mac(),
     password: {
-      args: ["passwordLenth"],
+      args: ["passwordLength"],
```
Comment (IvanGoncharov) [inline src/fake/functions/internet.ts]: @kristianmandrup Can you please extract it into separate PR.

**Comment by kristianmandrup** [comment]
I'd suggest you just go through it one commit at a time to follow the changes and my thinking.
Consult the updated `Changelog.md` for an overview of the main changes. I will test it shortly on my end.

Happy for now. Fully customisable!

**Comment by IvanGoncharov** [comment]
> I'd suggest you just go through it one commit at a time to follow the changes and my thinking.

@kristianmandrup Sorry, but every big change should be submitted and discussed as a separate PR. If you want your changes to be merged please break them into separate PRs.

**Comment by kristianmandrup** [comment]
Sure, but not until my work has stabilised and has been tested. I'll close this PR for now then.

**Comment by IvanGoncharov** [comment]
@kristianmandrup Great 👍 
Sure, take as much time as you need.
Looking forward to your PRs.

**Comment by kristianmandrup** [comment]
Sure. I will add a jest test suite as well to document it works as expected and make it clear how to use and extend the new functionality.
